plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/InMemoryRuleProcessingService.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.rule;
+
+import org.opensearch.plugin.wlm.rule.attribute_extractor.AttributeExtractor;
+import org.opensearch.plugin.wlm.rule.storage.AttributeValueStore;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This class is responsible for managing in-memory view of Rules and Find matching Rule for the request
+ */
+public class InMemoryRuleProcessingService {
+
+    /**
+     * Default constructor
+     */
+    public InMemoryRuleProcessingService() {}
+
+    /**
+     * Adds the rule to in-memory view
+     * @param rule to be added
+     */
+    public void add(final Rule rule) {
+        new AddRuleOperation(rule).perform();
+    }
+
+    /**
+     * Removes the rule from in-memory view
+     * @param rule to be removed
+     */
+    public void remove(final Rule rule) {
+        new DeleteRuleOperation(rule).perform();
+    }
+
+    /**
+     * Evaluates the label for the current request. It finds the matches for each attribute value and then it is an
+     * intersection of all the matches
+     * @param attributeExtractors list of extractors which are used to get the attribute values to find the
+     *                           matching rule
+     * @return a label if there is unique label otherwise empty
```

Comment:
What if there are multiple matches? Is that scenario possible?

Reply:
There shouldn't be multiple matches because it is a prefix structure and there will never be two different paths sharing a common prefix.